lib/ecl/ecl_grid.cpp

```diff
@@ -1533,6 +1534,14 @@ static ecl_grid_type * ecl_grid_alloc_empty(ecl_grid_type * global_grid,
                                             int nz,
                                             int lgr_nr,
                                             bool init_valid) {
+  /* Check for int overflow in a roundabout way */
```

Comment:
Why the grid size needs to be `< INT_MAT`?

Reply:
https://github.com/equinor/libecl/blob/f1443476bc784f5b1d052e33cf1ea3235c41a00a/python/tests/ecl_tests/test_grid.py#L129-L135

The `size` (an int) overflows in one of the test due to x = y = z = 2000, and 2000<sup>3</sup> > `INT_MAX`. This is undefined behaviour. Because `INT_MAX`<sup>3</sup> > `ULONG_MAX`, we can't simply cast each `int` to `uint64_t` and do the multiplication, so we have to test each multiplication in this roundabout way.

The previous version had a `size` that overflowed and then would `util_abort` when trying to malloc <strong>-</strong>589934592 bytes, so adding this check will make this more robust.

Reply:
We need to find out what's the maximum grid size in use

Reply:
> We need to find out what's the maximum grid size in use

The `.EGRID` file format dictates that the number of elements in the `ACTNUM` array can be represented with a 32bit signed integer - i.e. `nx*ny*nz < 2**31 - 1`